src/mscorlib/shared/System/IO/Path.Windows.cs

```diff
@@ -119,14 +119,21 @@ public static bool IsPathRooted(string path)
         {
             if (path != null)
             {
-                int length = path.Length;
-                if ((length >= 1 && PathInternal.IsDirectorySeparator(path[0])) ||
-                    (length >= 2 && PathInternal.IsValidDriveChar(path[0]) && path[1] == PathInternal.VolumeSeparatorChar))
-                    return true;
+                IsPathRooted(path.AsReadOnlySpan());
```

Comment:
you don't need AsReadOnlySpan, it can go through the string path, since you already have a string.